bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/TextLayout.java

```diff
@@ -2137,8 +2137,9 @@ public FontMetrics getLineMetrics (int lineIndex) {
 	OS.DeleteDC(srcHdc);
 	device.internal_dispose_GC(hDC, null);
 
-	int ascentInPoints = this.ascent;
-	int descentInPoints = this.descent;
+	final int zoom = getZoom();
+	int ascentInPoints =  Math.max(DPIUtil.scaleDown(getDevice(), lptm.tmAscent, zoom), this.ascent);
```

Comment:
Why not use the field `device` instead of calling `getDevice()` so often? I mean the call to `internal_new_GC` already checked if the device is disposed so starting in line 2132 you can safely assume the device is not disposed.

Reply:
Good point, Adapted all places in the changed method to use the field directly.